pkg/detector/detector.go

```diff
@@ -1365,9 +1378,12 @@ func (d *ResourceDetector) CleanupResourceTemplateMarks(objRef workv1alpha2.Obje
 }
 
 // CleanupResourceBindingMarks removes marks, such as labels and annotations, from resource binding.
-func (d *ResourceDetector) CleanupResourceBindingMarks(rb *workv1alpha2.ResourceBinding, cleanupFunc func(obj metav1.Object)) error {
+func (d *ResourceDetector) CleanupResourceBindingMarks(rb *workv1alpha2.ResourceBinding, cleanupFunc func(obj metav1.Object) bool) error {
 	return retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
-		cleanupFunc(rb)
+		if !cleanupFunc(rb) {
+			klog.Infof("The marks on resource binding %s/%s have changed, and no need to be cleaned up", rb.GetNamespace(), rb.GetName())
```

Comment:
`resource binding` => `ResourceBinding`

Reply:
I also think this might be better, but choosing to write it this way is to maintain consistency with
https://github.com/karmada-io/karmada/blob/b4b6d692ca963c3d53de4a61077f4380613be0a5/pkg/detector/detector.go#L1380

Reply:
don't mind it, of course, you can also change the others if it is convenient for you. :hugs:

Reply:
I expected to write like this: `resourceBinding(%s/%s)` ^-^
A matter of style, a thousand faces.